core/src/test/scala/org/apache/spark/scheduler/TaskSetManagerSuite.scala

```diff
@@ -819,6 +833,8 @@ class TaskSetManagerSuite extends SparkFunSuite with LocalSparkContext with Logg
       assert(sched.endedTasks(id) === Success)
     }
 
+    // check speculatable tasks requires a non-zero runtime for the tasks
```

Comment:
How about:

checkSpeculatableTasks checks that the task runtime is greater than the threshold for speculating.  Since we use a threshold of 0 for speculation, tasks need to be running for > 0ms, so advance the clock by 1ms here.